packages/legend-application/src/application/LegendApplication.ts

```diff
@@ -312,16 +312,39 @@ export abstract class LegendApplication {
     >,
   ): Promise<void>;
 
-  async start(): Promise<void> {
+  async start(providedConfig?: {
```

Comment:
I think we should do this differently. Let's refactor so `fetchApplicationConfiguration` does not take a `baseUrl` as a parameter. This can be done easily, because we can replace that with `this.baseUrl`.

Now, all you need to do if you need to pass in a custom config is to create a subclass of `LegendApplication` or `LegendStudioApplication` says `VSCodeLegendStudioApplication` and override the `fetchApplicationConfiguration` with what you need.